featurebyte/routes/historical_feature_table/controller.py

```diff
@@ -78,9 +78,16 @@ async def get_payload(
     async def get_validation_parameters(
         self, table_create: HistoricalFeatureTableCreate
     ) -> ValidationParameters:
-        # feature cluster group feature graph by feature store ID, only single feature store is
-        # supported
-        feature_cluster = table_create.featurelist_get_historical_features.feature_clusters[0]
+        # feature cluster group feature graph by feature store ID, only single feature store is supported
+
+        feature_clusters = table_create.featurelist_get_historical_features.feature_clusters
+        if not feature_clusters:
+            # feature_clusters has become optional, need to derive it from feature_list_id when it is not set
+            feature_clusters = await self.feature_list_service.get_feature_clusters(
+                table_create.featurelist_get_historical_features.feature_list_id  # type: ignore[arg-type]
+            )
+
+        feature_cluster = feature_clusters[0]
```

Comment:
will there always be one? or does it not matter if there are multiple? might be useful to add a comment, or add an assert on length here~